addons/slack/template/index.js

```diff
@@ -0,0 +1,58 @@
+const slack = require('./slack.js');
+const Sdk = require('@fusebit/add-on-sdk');
+const { WebClient } = require('@slack/web-api');
+const Superagent = require('superagent');
+
+module.exports = async (ctx) => {
+  const sdk = {
+    bot: new WebClient(ctx.configuration.slack_bot_access_token),
+    user: ctx.configuration.slack_user_access_token ? new WebClient(ctx.configuration.slack_user_access_token) : null,
+  };
+
+  // TODO What prevents someone who finds out this URL to
+  // send unauthorized events?
+
+  if (ctx.url.endsWith('events/ingest')) {
+    Sdk.debug('Received Events API event');
+
+    let response;
+    try {
+      if (ctx.method === 'POST') {
+        let event = ctx.body;
+        Sdk.debug('Valid event', event);
+
+        // Attach all event handlers off a nested `slack` property
+        let handlers = slack.slack;
+        if (handlers) {
+          let functions = Object.getOwnPropertyNames(handlers).filter((p) => typeof handlers[p] === 'function');
+          let toInvoke = functions.filter((f) => f == event.type || f == 'all');
+
+          Sdk.debug('Invoking handlers', toInvoke);
+          toInvoke.forEach((t) => {
+            handlers[t](sdk, event, {
```

Comment:
Shouldn't handlers be async, and this code `await handlers[t](...)`?

Reply:
They can be async, but do you actually want to await them here or just return as fast as possible? That was the intent

Reply:
In a general case the handlers should be async to allow for asynchronous work to happen (e.g. calling back to APIs) - I would require that the handlers are async functions (i.e. return a promise). Then, I would call `Promise.all` on them in parallel - I've seen you use this pattern elsewhere. 

And yes, we need to wait for them to complete, otherwise Lambda my freeze the execution environment before they finish asynchronously.

Reply:
Fixed